### DISCUSSION TITLE
How to attach Slint to an external window (parent HWND on Win32 / NSView on macOS)

### DISCUSSION THREAD

**Discussion by basdp** [slint-ui/slint#6585] [Q&A]
Now that Slint is very vocal about it's fitness for supporting Audio Plugin development, we need documentation and examples to support that. I've tried piecing things together but have had no luck in 'embedding' an Slint view into an external parent HWND on Windows or a NSView on macOS. There is this C++ only example for Windows, which is not very useful as my whole reason to go with Rust is to get away from C++ and it targets Skia (which is not the best option for this). 

Can anybody give me some pointers in Rust how to attach the Slint view?

**Comment by ogoffart** [comment]
Maybe you'll find this project interesting: https://github.com/ilmai/plugin-things

**Reply by basdp** [reply]
That's really interesting, it looks like it fits my needs perfectly. 

I think we should extract a minimal example of the code to attach to an external window to include in the Slint examples here. I might be able to work that out in the coming days.

Do you have any pointers to how to use 'plugin-things'? There is no documentation or an example plugin right?

**Reply by asuper0** [reply]
I don't know how does it works ... It seems that the `OsWindow` should do a lot to handle UI event. Hope to see some document about it.
